Assets/Scripts/Editor/FaceAnimationGenerator.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEditor.Animations;
+using UnityEngine;
+
+/// <summary>
+/// Creates a new tab under Tools/Dialog System called
+/// Face Animation Generator. This tab opens up a new
+/// Editor Window for the user to type in the name
+/// of their new character, prefably using spaces
+/// for when the name needs to split up into multiple
+/// words. Below the name input field is an expandable
+/// list of sprites and Names. The name value is to hold
+/// the name of the expression such as mad, sad,glad,neutral,etc...
+/// After the button is pressed, a new folder is created
+/// with the characters name under "Assets/Animations/Portraits".
+/// Inside is the new animation clips generated.
+/// And the Portrait Animator controller ,which is assumed
+/// to be located at Assets/Animations/Portaits/PortraitAnimator.controller,
+/// automatically has the new clip added. 
+/// </summary>
+public class FaceAnimationGenerator : EditorWindow
+{
```

Comment:
This is really cool. I haven't done a lot of programming with the Unity editor myself, so I really enjoyed looking at how this was done. Awesome job and thanks for contributing it! 🙂